book/src/adding_languages.md

```diff
@@ -0,0 +1,51 @@
+# Adding languages
+
+## Submodules
+
+To add a new langauge, you should first add a tree-sitter submodule. To do this, you can run the command
+```sh
+$ git submodule add -f <repository> helix-syntax/languages/tree-sitter-<name>
+```
+For example, to add tree-sitter-ocaml you would run
+```sh
+$ git submodule add -f https://github.com/tree-sitter/tree-sitter-ocaml helix-syntax/languages/tree-sitter-ocaml
+```
+Make sure the submodule is shallow by doing
+```sh
+git config -f .gitmodules submodule.helix-syntax/languages/tree-sitter-<name>.shallow true
+```
+You can also edit the `.gitmodules` file.
+
+## languages.toml
+
+Next, you need to add the language to `languages.toml` found in the root of the repository. The `languages.toml` file is included at compilation time.
+
+These are the available keys and descriptions for the file.
+
+| Key           | Description                                                   |
+| ----          | -----------                                                   |
+| name          | The name of the language                                      |
+| scope         | I don't know what this does                                   |
```

Comment:
https://tree-sitter.github.io/tree-sitter/syntax-highlighting#basics



scope (required) - A string like "source.js" that identifies the language. Currently, we strive to match the scope names used by popular TextMate grammars and by the Linguist library.

Reply:
Usually `source.<name>` or `text.<name>` in case of markup languages

Reply:
What is the difference between source and text?

Reply:
https://www.sublimetext.com/docs/scope_naming.html#text